### ISSUE TITLE
Uncaught TypeError: listener.close is not a function

### ISSUE THREAD

**Issue by robhuska** [icetee/remote-ftp#378]
Any time I save a partial scss file that is compiled into a watched css file I get this error. It only recently started. Yesterday however, I did make a change that fixed issue #366 which was in this pull request #367. The issue did not start right away, and it was not every file, now it is every file I save. Everything saves ok and is uploaded, I just get the error as well.

**Atom Version**: 1.7.3
**System**: Mac OS X 10.11.4
**Thrown From**: [Remote-FTP](https://github.com/mgrenier/remote-ftp) package, v0.8.3
### Stack Trace

Uncaught TypeError: listener.close is not a function

```
At /Users/rhuska/.atom/packages/Remote-FTP/lib/client.js:787

TypeError: listener.close is not a function
    at remove (/Users/rhuska/.atom/packages/Remote-FTP/lib/client.js:787:16)
    at Client.module.exports.Client.watch.removeListeners (/Users/rhuska/.atom/packages/Remote-FTP/lib/client.js:813:5)
    at FSWatcher.<anonymous> (/Users/rhuska/.atom/packages/Remote-FTP/lib/client.js:756:36)
    at emitTwo (events.js:87:13)
    at FSWatcher.emit (events.js:172:7)
    at FSEvent.FSWatcher._handle.onchange (fs.js:1310:12)
```
### Commands

```
  2x -9:55.2.0 core:move-right (atom-text-editor.editor.is-focused)
     -9:54.7.0 core:select-right (atom-text-editor.editor.is-focused)
     -9:54.2.0 core:save (atom-text-editor.editor.is-focused)
     -9:36.1.0 core:select-left (atom-text-editor.editor.is-focused)
     -9:33.5.0 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
     -9:33 core:save (atom-text-editor.editor.is-focused)
     -9:18.9.0 core:copy (atom-text-editor.editor.is-focused)
     -9:17.1.0 core:paste (atom-text-editor.editor.is-focused)
     -9:16.8.0 core:save (atom-text-editor.editor.is-focused)
     -7:12 core:select-right (atom-text-editor.editor.is-focused)
     -7:09.7.0 core:backspace (atom-text-editor.editor.is-focused)
  4x -7:09.4.0 core:save (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "SFTP-deployment",
      "remote-sync",
      "wordpress-dictionary",
      "autocomplete-wordpress-hooks",
      "atom-wordpress"
    ],
    "packagesWithKeymapsDisabled": [],
    "themes": [
      "atom-material-ui",
      "atom-material-syntax-dark"
    ]
  },
  "Remote-FTP": {
    "autoUploadOnSave": "only when connected"
  }
}
```
### Installed Packages

``` coffee
# User
Remote-FTP, v0.8.3
atom-commander, v0.7.0
atom-material-syntax-dark, v0.2.5
atom-material-ui, v1.2.11
autoclose-html, v0.23.0
file-icons, v1.7.11
highlight-line, v0.11.1
javascript-snippets, v1.2.1
minimap, v4.23.5
minimap-cursorline, v0.1.0
minimap-pigments, v0.2.1
project-manager, v2.9.7
sync-settings, v0.7.2
tasks, v2.6.0
terminal-plus, v0.14.5
wordpress-api, v1.2.2

# Dev
No dev packages
```

**Comment by robhuska** [comment]
I noticed if I disconnect, then just save my '.ftpconfig' file and then reconnect, the issue goes away for a while, but it always comes back.

**Comment by robhuska** [comment]
Another thing I noticed, it seems to have something to do with my sourcemaps. I had also received the error that my source map was deleted or did not exist and it was being removed from my watch list. but the file was there. So I deleted it and the file regenerated on my next save. When I do this, the error goes away, but again it always comes back eventually.

Please help, this issue is very persistent! It is actually making working in Atom kind of difficult since the error takes up so much space in the editor...

**Comment by jimmaaay** [comment]
I believe if you change `listener.close()` to `listener[0].close()` in `/lib/client.js` on line 786 your problem should be fixed. I will create a pull request fixing this if no one else does in a few days time.

I do notice this a lot more with soucemaps I believe it has something to do with the way they are edited, as I can replicate the error by editing a watched file with text edit on mac. For some reason this fires the rename event on the watcher, which then in the current implementation removes the watcher, or at least attempts to. 

``` javascript
            function remove() {
                if (fileName) {

                    for (var i = 0; i < self.watch.files.length; i++) {
                        var file = self.watch.files[i];
                        if (file != fileName) continue;
                        var listener = self.watch.listeners.splice(i, 1);
                        listener[0].close();
                        self.watch.listenerCount--;
                        self.watch.files.splice(i, 1);
                    }

                } else {
```

**Comment by jpxd** [comment]
Fixed with #446
